### PR TITLE
testing: dont skip test_ops suite for operators testing against scipy

### LINKED ISSUE #54152: erf and erfc tests in test_ops.py sometimes don't exist
In this CI job you can see that erf/erfc have failed: https://app.circleci.com/pipelines/github/pytorch/pytorch/286464/workflows/f98f8828-6010-404e-bfcb-8dcbdb68c4ee/jobs/11609202

When I run the test locally, no such test exists

```
$ python test/test_ops.py TestCommonMETA.test_variant_consistency_eager_erf_meta_int32
E
======================================================================
ERROR: test_variant_consistency_eager_erf_meta_int32 (unittest.loader._FailedTest)
----------------------------------------------------------------------
AttributeError: type object 'TestCommonMETA' has no attribute 'test_variant_consistency_eager_erf_meta_int32'

----------------------------------------------------------------------
Ran 1 test in 0.000s
```

I'm guessing this is build configuration related. But the set of defined tests should NEVER vary based on build config; we should always define the test (and skip it if it's not necessary!)

cc @mruberry @VitalyFedyunin @walterddr

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #54152

### CODE REVIEW

File: test/test_unary_ufuncs.py [hunk 1]
```diff
@@ -25,6 +25,8 @@
 if TEST_SCIPY:
     import scipy
 
+reference_filtered_ops = list(filter(lambda op: op.ref is not _NOTHING, unary_ufuncs))
```
Comment (mruberry) [inline test/test_unary_ufuncs.py:28]: Add a comment explaining this filter

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3059,181 +3108,148 @@ def _make_tensor_helper(shape, low=None, high=None):
...
+                                                  torch.bfloat16: 1e-2}),),
+                   skips=(
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_extremal',
+                                device_type='cpu', dtypes=[torch.cfloat, torch.cdouble]),
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_hard',
+                                device_type='cpu', dtypes=[torch.cfloat, torch.cdouble]),
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics_normal',
+                                device_type='cpu', dtypes=[torch.cfloat, torch.cdouble]),
+                       # RuntimeError: sigmoid does not support automatic differentiation for outputs with complex dtype.
+                       SkipInfo('TestCommon', 'test_variant_consistency_jit',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3136]: These skips can be removed, but new skips or metadata will need to be added. We have metadata for whether an op supports complex autograd that can be used here, I think.

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3059,181 +3108,148 @@ def _make_tensor_helper(shape, low=None, high=None):
...
+                                dtypes=[torch.complex64, torch.complex128]),
+                       SkipInfo('TestCommon', 'test_variant_consistency_eager',
+                                dtypes=[torch.complex64, torch.complex128]),),
+                   dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16),
+                   dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.bfloat16),
+                   dtypesIfCUDA=all_types_and(torch.bool, torch.half, torch.bfloat16),
+                   safe_casts_outputs=True,
+                   assert_autodiffed=True,
+                   test_complex_grad=False),  # Reference: https://github.com/pytorch/pytorch/issues/48552
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3145]: `supports_complex_autograd=False`

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -1607,6 +1607,55 @@ def _make_tensor_helper(shape, low=None, high=None):
     return samples
 
 
+def scipy_reference_wrapper(ref):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1610]: Clarify how and when to use this.

"If you're using a SciPy op as a reference, then pass it wrapped in a lambda to this function, which will either return the lambda or, if SciPy is unavailable, _NOTHING. Wrapping the operation in a lambda will prevent the Python interpreter from attempting to discover the operation, which will fail if SciPy is unavailable. See the OpInfo for digamma for an example."

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -3059,181 +3108,148 @@ def _make_tensor_helper(shape, low=None, high=None):
...
+                       SkipInfo('TestCommon', 'test_variant_consistency_eager',
+                                dtypes=[torch.complex64, torch.complex128]),),
+                   dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16),
+                   dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.bfloat16),
+                   dtypesIfCUDA=all_types_and(torch.bool, torch.half, torch.bfloat16),
+                   safe_casts_outputs=True,
+                   assert_autodiffed=True,
+                   test_complex_grad=False),  # Reference: https://github.com/pytorch/pytorch/issues/48552
+    UnaryUfuncInfo('digamma',
+                   ref=scipy_reference_wrapper(lambda x: scipy.special.digamma(x)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3147]: This lambda is perfect, but if this PR does want to cite it as an example then:

```
lambda *args, **kwargs: scipy.special.digamma(*args, **kwargs)
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 6]
```diff
@@ -3059,181 +3108,148 @@ def _make_tensor_helper(shape, low=None, high=None):
...
+                   decorators=(precisionOverride({torch.float16: 5e-1}),),
+                   dtypes=all_types_and(torch.bool),
+                   dtypesIfCPU=all_types_and(torch.bool),
+                   dtypesIfCUDA=all_types_and(torch.bool, torch.half),
+                   skips=(
+                       # In some cases, output is NaN (for input close to
+                       # negative integers) especially due to reduced precision
+                       # in float16 and NaN's can't be tested for equality.
+                       SkipInfo('TestCommon', 'test_variant_consistency_jit',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3156]: This skip is not needed but a new skip may be

**Review by mruberry — APPROVED** [review]
Nice work, @kshitij12345. I made a few suggestions. The biggest thing is that this PR will need a rebase to account for the latest OpInfo changes. Once that happens I'll prioritize its landing since it's a larger PR.

File: torch/testing/_internal/common_methods_invocations.py (hunk 6 above)
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py]: Have pulled the latest master and just update the `ref` arguments with wrapped. Kept the other stuff as is in the master.

File: torch/testing/_internal/common_methods_invocations.py (hunk 5 above)
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py]: Have only updated the lambda for `digamma` (being the example). 
Have kept other lambda's more closer to their original signature

**Comment by kshitij12345** [comment]
Have mostly addressed the reviews. Hopefully should be good to go. PTAL :)

**Pull request closed without merging**